In [1]:
import pandas as pd


In [4]:
df=pd.read_csv('../alzheimers_disease_data.csv')
df

PatientID  Age  Gender  Ethnicity  EducationLevel        BMI  Smoking  \
0          4751   73       0          0               2  22.927749        0   
1          4752   89       0          0               0  26.827681        0   
2          4753   73       0          3               1  17.795882        0   
3          4754   74       1          0               1  33.800817        1   
4          4755   89       0          0               0  20.716974        0   
...         ...  ...     ...        ...             ...        ...      ...   
2144       6895   61       0          0               1  39.121757        0   
2145       6896   75       0          0               2  17.857903        0   
2146       6897   77       0          0               1  15.476479        0   
2147       6898   78       1          3               1  15.299911        0   
2148       6899   72       0          0               2  33.289738        0   

      AlcoholConsumption  PhysicalActivity  DietQuality  ...  \
0              13.297218          6.327112     1.347214  ...   
1               4.542524          7.619885     0.518767  ...   
2              19.555085          7.844988     1.826335  ...   
3              12.209266          8.428001     7.435604  ...   
4              18.454356          6.310461     0.795498  ...   
...                  ...               ...          ...  ...   
2144            1.561126          4.049964     6.555306  ...   
2145           18.767261          1.360667     2.904662  ...   
2146            4.594670          9.886002     8.120025  ...   
2147            8.674505          6.354282     1.263427  ...   
2148            7.890703          6.570993     7.941404  ...   

      MemoryComplaints  BehavioralProblems       ADL  Confusion  \
0                    0                   0  1.725883          0   
1                    0                   0  2.592424          0   
2                    0                   0  7.119548          0   
3                    0                   1  6.481226          0   
4                    0                   0  0.014691          0   
...                ...                 ...       ...        ...   
2144                 0                   0  4.492838          1   
2145                 0                   1  9.204952          0   
2146                 0                   0  5.036334          0   
2147                 0                   0  3.785399          0   
2148                 0                   1  8.327563          0   

      Disorientation  PersonalityChanges  DifficultyCompletingTasks  \
0                  0                   0                          1   
1                  0                   0                          0   
2                  1                   0                          1   
3                  0                   0                          0   
4                  0                   1                          1   
...              ...                 ...                        ...   
2144               0                   0                          0   
2145               0                   0                          0   
2146               0                   0                          0   
2147               0                   0                          0   
2148               1                   0                          0   

      Forgetfulness  Diagnosis  DoctorInCharge  
0                 0          0       XXXConfid  
1                 1          0       XXXConfid  
2                 0          0       XXXConfid  
3                 0          0       XXXConfid  
4                 0          0       XXXConfid  
...             ...        ...             ...  
2144              0          1       XXXConfid  
2145              0          1       XXXConfid  
2146              0          1       XXXConfid  
2147              1          1       XXXConfid  
2148              1          0       XXXConfid  

[2149 rows x 35 columns]

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2149 entries, 0 to 2148
Data columns (total 35 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   PatientID                  2149 non-null   int64  
 1   Age                        2149 non-null   int64  
 2   Gender                     2149 non-null   int64  
 3   Ethnicity                  2149 non-null   int64  
 4   EducationLevel             2149 non-null   int64  
 5   BMI                        2149 non-null   float64
 6   Smoking                    2149 non-null   int64  
 7   AlcoholConsumption         2149 non-null   float64
 8   PhysicalActivity           2149 non-null   float64
 9   DietQuality                2149 non-null   float64
 10  SleepQuality               2149 non-null   float64
 11  FamilyHistoryAlzheimers    2149 non-null   int64  
 12  CardiovascularDisease      2149 non-null   int64  
 13  Diabetes                   2149 non-null   int64

In [7]:
 df.iloc[:, 1:16].describe()


Age       Gender    Ethnicity  EducationLevel          BMI  \
count  2149.000000  2149.000000  2149.000000     2149.000000  2149.000000   
mean     74.908795     0.506282     0.697534        1.286645    27.655697   
std       8.990221     0.500077     0.996128        0.904527     7.217438   
min      60.000000     0.000000     0.000000        0.000000    15.008851   
25%      67.000000     0.000000     0.000000        1.000000    21.611408   
50%      75.000000     1.000000     0.000000        1.000000    27.823924   
75%      83.000000     1.000000     1.000000        2.000000    33.869778   
max      90.000000     1.000000     3.000000        3.000000    39.992767   

           Smoking  AlcoholConsumption  PhysicalActivity  DietQuality  \
count  2149.000000         2149.000000       2149.000000  2149.000000   
mean      0.288506           10.039442          4.920202     4.993138   
std       0.453173            5.757910          2.857191     2.909055   
min       0.000000            0.002003          0.003616     0.009385   
25%       0.000000            5.139810          2.570626     2.458455   
50%       0.000000            9.934412          4.766424     5.076087   
75%       1.000000           15.157931          7.427899     7.558625   
max       1.000000           19.989293          9.987429     9.998346   

       SleepQuality  FamilyHistoryAlzheimers  CardiovascularDisease  \
count   2149.000000              2149.000000            2149.000000   
mean       7.051081                 0.252210               0.144253   
std        1.763573                 0.434382               0.351428   
min        4.002629                 0.000000               0.000000   
25%        5.482997                 0.000000               0.000000   
50%        7.115646                 0.000000               0.000000   
75%        8.562521                 1.000000               0.000000   
max        9.999840                 1.000000               1.000000   

          Diabetes   Depression   HeadInjury  
count  2149.000000  2149.000000  2149.000000  
mean      0.150768     0.200558     0.092601  
std       0.357906     0.400511     0.289940  
min       0.000000     0.000000     0.000000  
25%       0.000000     0.000000     0.000000  
50%       0.000000     0.000000     0.000000  
75%       0.000000     0.000000     0.000000  
max       1.000000     1.000000     1.000000

In [8]:
df.iloc[:, 16:35].describe()


Hypertension   SystolicBP  DiastolicBP  CholesterolTotal  \
count   2149.000000  2149.000000  2149.000000       2149.000000   
mean       0.148906   134.264774    89.847836        225.197519   
std        0.356079    25.949352    17.592496         42.542233   
min        0.000000    90.000000    60.000000        150.093316   
25%        0.000000   112.000000    74.000000        190.252963   
50%        0.000000   134.000000    91.000000        225.086430   
75%        0.000000   157.000000   105.000000        262.031657   
max        1.000000   179.000000   119.000000        299.993352   

       CholesterolLDL  CholesterolHDL  CholesterolTriglycerides         MMSE  \
count     2149.000000     2149.000000               2149.000000  2149.000000   
mean       124.335944       59.463533                228.281496    14.755132   
std         43.366584       23.139174                101.986721     8.613151   
min         50.230707       20.003434                 50.407194     0.005312   
25%         87.195798       39.095698                137.583222     7.167602   
50%        123.342593       59.768237                230.301983    14.441660   
75%        161.733733       78.939050                314.839046    22.161028   
max        199.965665       99.980324                399.941862    29.991381   

       FunctionalAssessment  MemoryComplaints  BehavioralProblems  \
count           2149.000000       2149.000000         2149.000000   
mean               5.080055          0.208004            0.156817   
std                2.892743          0.405974            0.363713   
min                0.000460          0.000000            0.000000   
25%                2.566281          0.000000            0.000000   
50%                5.094439          0.000000            0.000000   
75%                7.546981          0.000000            0.000000   
max                9.996467          1.000000            1.000000   

               ADL    Confusion  Disorientation  PersonalityChanges  \
count  2149.000000  2149.000000     2149.000000         2149.000000   
mean      4.982958     0.205212        0.158213            0.150768   
std       2.949775     0.403950        0.365026            0.357906   
min       0.001288     0.000000        0.000000            0.000000   
25%       2.342836     0.000000        0.000000            0.000000   
50%       5.038973     0.000000        0.000000            0.000000   
75%       7.581490     0.000000        0.000000            0.000000   
max       9.999747     1.000000        1.000000            1.000000   

       DifficultyCompletingTasks  Forgetfulness    Diagnosis  
count                2149.000000    2149.000000  2149.000000  
mean                    0.158678       0.301536     0.353653  
std                     0.365461       0.459032     0.478214  
min                     0.000000       0.000000     0.000000  
25%                     0.000000       0.000000     0.000000  
50%                     0.000000       0.000000     0.000000  
75%                     0.000000       1.000000     1.000000  
max                     1.000000       1.000000     1.000000

In [9]:
df.columns

Index(['PatientID', 'Age', 'Gender', 'Ethnicity', 'EducationLevel', 'BMI',
       'Smoking', 'AlcoholConsumption', 'PhysicalActivity', 'DietQuality',
       'SleepQuality', 'FamilyHistoryAlzheimers', 'CardiovascularDisease',
       'Diabetes', 'Depression', 'HeadInjury', 'Hypertension', 'SystolicBP',
       'DiastolicBP', 'CholesterolTotal', 'CholesterolLDL', 'CholesterolHDL',
       'CholesterolTriglycerides', 'MMSE', 'FunctionalAssessment',
       'MemoryComplaints', 'BehavioralProblems', 'ADL', 'Confusion',
       'Disorientation', 'PersonalityChanges', 'DifficultyCompletingTasks',
       'Forgetfulness', 'Diagnosis', 'DoctorInCharge'],
      dtype='object')

### Interpretação dos dados 


Após analisar as métricas de cada coluna, é possível compreender melhor o comportamento e as características predominantes na amostra. A seguir, apresentamos as interpretações baseadas nos valores médios e nos desvios padrão de cada variável, proporcionando uma visão mais clara sobre a frequência e a variabilidade de cada atributo analisado.

- PatientID (Identificação do Paciente) : Esta coluna serve exclusivamente para identificar de forma única cada paciente no dataset. Não possui relevância estatística direta, mas é fundamental para rastrear informações específicas dos indivíduos.

- Age (Idade): A idade média dos pacientes é de 74 anos, com um desvio padrão de aproximadamente 9 anos, o que indica uma variação considerável entre as idades. A faixa etária abrange desde pacientes com 60 anos (mínimo) até 90 anos (máximo). Isso sugere que o estudo está focado em um público idoso, faixa etária geralmente mais suscetível a doenças neurodegenerativas, como Alzheimer.

- Gender (Gênero): Esta coluna é binária, onde 0 representa masculino e 1 representa feminino. O valor médio de 0,5 sugere uma distribuição equilibrada entre os gêneros, indicando que ambos os sexos estão igualmente representados na amostra.

- Ethnicity (Etnia): Os valores desta coluna indicam as categorias de etnia dos pacientes, com 0 para Caucasian (caucasianos) e 1 para African American (afro-americanos). A média de 0,69 sugere que a maioria dos pacientes pertence às categorias 0 e 1, com predominância leve de afro-americanos. Esse dado pode ser relevante para estudar variações epidemiológicas entre diferentes grupos étnicos.

- EducaionLevel (Nível Educacional): Representa o nível educacional dos pacientes, onde níveis mais altos indicam maior escolaridade.
Os resultados indicam que, em média, os participantes completaram o ensino médio. Esse dado é relevante porque estudos mostram que níveis mais altos de escolaridade podem estar associados a uma maior reserva cognitiva, o que pode influenciar a progressão de doenças como Alzheimer.

- BMI (IMC): O IMC médio indica que a maioria dos pacientes está na categoria de sobrepeso. Há pacientes em extremos de peso, com valores mínimos indicando baixo peso (IMC de 15) e máximos chegando à obesidade grau II (IMC de 39,99). Essas condições podem influenciar negativamente a saúde geral e cognitiva dos pacientes.

- Smoking (Fumo): Esta coluna é binária, onde 0 significa não fumante e 1 significa fumante. A média de 0,28 indica que cerca de 28% dos pacientes possuem o hábito de fumar. Este dado é relevante, pois o tabagismo é um fator de risco para diversas condições de saúde, incluindo declínio cognitivo.

- AlcoholConsumption (Consumo de Álcool): O consumo médio de álcool por paciente é de 10 unidades por semana, com uma variação de 5 unidades para mais ou para menos. Isso sugere que há uma parcela considerável de pacientes com consumo moderado a alto de álcool, fator que pode impactar a saúde cerebral e o sistema nervoso.

- PhysicalActivity (Atividade Física): A prática média de 4 horas semanais indica que a maioria dos pacientes não atinge níveis de atividade física considerados ideais. A variação sugere que alguns pacientes realizam cerca de 1 hora diária de atividade física, enquanto outros são completamente sedentários. A atividade física é um fator importante para a saúde geral, especialmente em populações idosas.

- DietQuality (Qualidade da Dieta): Com uma pontuação média de 4,99, a dieta dos pacientes é classificada como razoável, com espaço significativo para melhorias. Uma alimentação equilibrada pode ter um impacto positivo na prevenção de doenças neurodegenerativas.

- SleepQuality (Qualidade do Sono): A qualidade média do sono é boa, considerando que a pontuação está acima de 7. Entretanto, a variação indica que alguns pacientes apresentam dificuldades, fator que pode estar associado ao declínio cognitivo.

- FamilyHistoryAlzheimers (Histórico Familiar de Alzheimer): Cerca de 25% dos pacientes possuem histórico familiar de Alzheimer, fator de risco importante para o desenvolvimento da doença.

- CardiovascularDisease (Doença Cardiovascular): Apenas 14% dos pacientes apresentam histórico de doenças cardiovasculares. Apesar de ser uma porcentagem baixa, é um fator relevante, pois doenças cardiovasculares podem influenciar a saúde cognitiva.

- Diabetes: Cerca de 15% dos pacientes possuem diagnóstico de diabetes, uma condição que pode estar associada a outros fatores de saúde relevantes no contexto do estudo.

- Depression (Depressão): Cerca de 20% dos pacientes apresentam depressão, o que indica uma presença significativa dessa condição entre os participantes.

- HeadInjury (Lesão na Cabeça): Apenas 9% dos pacientes relataram histórico de lesões na cabeça, representando uma minoria na amostra.

- Hypertension (Hipertensão): Apenas 14% dos pacientes possuem hipertensão, mas é um fator que deve ser monitorado, pois pode estar relacionado a complicações vasculares que afetam a saúde cerebral.

- SystolicBP (Pressão Sistólica): A média da pressão sistólica dos pacientes está acima do valor considerado ideal (<120 mmHg), o que pode indicar a necessidade de atenção em relação aos níveis de pressão arterial.

- DiastolicBP (Pressão Diastólica): Os valores médios da pressão diastólica também estão acima do limite recomendado (<80 mmHg), reforçando a relevância de monitorar essa variável.

- CholeserolTotal (Colesterol Total): O colesterol total médio está elevado, considerando que valores normais são inferiores a 200 mg/dL.

- CholesterolLDL (Colesterol LDL):Os níveis médios de LDL ("colesterol ruim") estão acima do recomendado, que é inferior a 100 mg/dL.

- CholeserolHDL (Colesterol HDL): Os níveis médios de HDL ("colesterol bom") estão dentro do intervalo ideal, geralmente acima de 40 mg/dL.

- CholeserolTriglycerides (Triglicerídeos): Os triglicerídeos apresentam uma média elevada, acima do limite recomendado de 150 mg/dL.

- MMSE (Mini-Mental State Examination): A pontuação média no MMSE sugere comprometimentos cognitivos, considerando que valores abaixo de 24 indicam possíveis alterações.

- FunctionalAssessment (Avaliação Funcional): Os escores médios indicam dificuldades moderadas nas atividades avaliadas, sugerindo impactos na funcionalidade dos pacientes.

- MemoryComplaints (Queixas de Memória): Cerca de 20% dos pacientes relatam queixas de memória, o que corresponde a uma parcela significativa.

- BehavioralProblems (Problemas Comportamentais): Apenas 15% dos pacientes relataram problemas comportamentais, indicando que essa condição não é muito frequente na amostra.

- ADL (Atividades de Vida Diária): A pontuação média indica que muitos pacientes apresentam dificuldades nas atividades de vida diária.

- Confusion (Confusão): Cerca de 20% dos pacientes apresentam confusão, o que pode indicar alguma dificuldade em processar informações ou situações.

- Disorientation (Desorientação): A desorientação foi relatada por 15% dos pacientes, representando uma parcela moderada da amostra.

- PersonalityChanges (Mudanças de Personalidade): Cerca de 15% dos pacientes relataram mudanças de personalidade, sugerindo que essa característica ocorre em menor frequência.

- DifficultyCompletingTasks (Dificuldade em Completar Tarefas): A dificuldade em completar tarefas foi identificada em 15% dos pacientes, sendo uma ocorrência moderada.

- Forgetfulness (Esquecimento): O esquecimento foi relatado por 30% dos pacientes, sendo uma das condições mais frequentes nesta amostra.

- Diagnosis (Diagnostico de Alzheimer): Cerca de 35% dos pacientes possuem diagnóstico de Alzheimer, indicando uma representatividade importante dessa condição na amostra.
